kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,284 @@
+//! For now we just use simple functions to deserialize table properties from strings. This allows
+//! us to relatively simply implement the functionality described in the protocol and expose
+//! 'simple' types to the user in the [`TableProperties`] struct. E.g. we can expose a `bool`
+//! directly instead of a `BoolConfig` type that we implement `Deserialize` for.
+
+use crate::expressions::ColumnName;
+
+use std::time::Duration;
+
+use serde::de::{self, Deserializer};
+use serde::Deserialize;
+
+const SECONDS_PER_MINUTE: u64 = 60;
+const SECONDS_PER_HOUR: u64 = 60 * SECONDS_PER_MINUTE;
+const SECONDS_PER_DAY: u64 = 24 * SECONDS_PER_HOUR;
+const SECONDS_PER_WEEK: u64 = 7 * SECONDS_PER_DAY;
+
+/// Transparently pass through the option deserialization to the inner type.
+pub(crate) fn deserialize_option<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    T::deserialize(deserializer).map(Some)
+}
+
+/// Deserialize a string representing a positive integer into an `Option<u64>`.
+pub(crate) fn deserialize_pos_int<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    let n: u64 = s.parse().map_err(de::Error::custom)?;
```

Comment:
Java doesn't even _have_ a u64 type. Should we deserialize as i64 instead and assert it's > 0? Otherwise, we could end up accepting (or even producing) values that java kernel could not handle.

Reply:
ah okay good catch will do